plugins/azure/resoto_plugin_azure/azure_client.py

```diff
@@ -66,8 +70,19 @@ def list(self, spec: AzureApiSpec, **kwargs: Any) -> List[Json]:
             else:
                 raise e
 
-    def delete(self, resource_id: str) -> None:
-        self.client.resources.delete_by_id(resource_id)
+    def delete(self, resource_id: str) -> bool:
+        try:
+            self._delete_by_id(resource_id, api_version="2021-04-01")
+        except HttpResponseError as e:
+            if e.error and e.error.code == "ResourceNotFoundError":
+                return False
+            else:
+                raise e
+
+        return True
+
+    def _delete_by_id(self, resource_id: str, api_version: str) -> None:
+        self.client.resources._delete_by_id_initial(resource_id=resource_id, api_version=api_version)
```

Comment:
What is the benefit of this method instead of `delete_by_id`?

Reply:
It is same method from official Azure API.